.github/workflows/publish.yml

```diff
@@ -24,11 +24,18 @@ jobs:
         run: |
           make build
 
+      - name: Retrieve version
+        id: version
+        run: |
+          version=$(cat version)
+          major=$(echo $version | cut -d. -f1)
+          echo "::set-output name=version::$version"
+          echo "::set-output name=major::$major"
```

Comment:
:monocle_face: Mmmm, I don't quite agree.

First, because the version, despite being semantic, can include the special value `next`. In your version, it will produce an error. But we should treat it as a major version:

```sh
$ version=next; echo $version | cut -d. -f1
next
```

Furthermore, the version is under the control of the CI and specifically under the control of the [semantic release](https://github.com/semantic-release), meaning that: on one hand, the file declaring it must exist, and on the other hand, its format is guaranteed to be compliant. So, I suggest we adhere to Bjarne Stroustrup's wise advice: "Make Simple Tasks Simple!" :relieved: